src/libs/actions/IOU.js

```diff
@@ -122,82 +139,266 @@ function createIOUTransaction(params) {
 }
 
 /**
- * Creates IOUSplit Transaction
+ * @param {Array} participants
+ * @param {String} currentUserLogin
+ * @param {Int} amount
+ * @param {String} comment
+ * @param {String} currency
+ * @param {String} locale
  *
- * @param {Object} params
- * @param {Array} params.splits
- * @param {String} params.comment
- * @param {Number} params.amount
- * @param {String} params.currency
+ * @return {Object}
  */
-function createIOUSplit(params) {
-    startLoadingAndResetError();
+function createSplitsAndOnyxData(participants, currentUserLogin, amount, comment, currency, locale) {
```

Comment:
I think this method would deserve a longer explanation of how exactly the Splits are achieved, what is done. It is quite dense and I think adding more comments to break it up would help with readability ad well as a longer JSDoc which will outline what the function does would be helpful

Reply:
Added a comment!